src/ServiceControl.Transports.PostgreSql/DatabaseDetails.cs

```diff
@@ -0,0 +1,144 @@
+﻿namespace ServiceControl.Transports.PostgreSql
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Npgsql;
+
+    public class DatabaseDetails
+    {
+        readonly string connectionString;
+
+        public string DatabaseName { get; }
+
+        public DatabaseDetails(string connectionString)
+        {
+            try
+            {
+                var builder = new NpgsqlConnectionStringBuilder
+                {
+                    ConnectionString = connectionString
+                };
+                DatabaseName = builder.Database;
+                this.connectionString = builder.ToString();
+            }
+            catch (Exception ex) when (ex is FormatException or ArgumentException)
+            {
+                throw new Exception("PostgreSQL Connection String could not be parsed.", ex);
+            }
+        }
+
+        public Task<string> TestConnection(CancellationToken cancellationToken)
+        {
+            try
+            {
+                return GetPostgreSqlVersion(cancellationToken);
+            }
+            catch (NpgsqlException ex) when (IsConnectionOrLoginIssue(ex))
+            {
+                throw new Exception($"Could not connect to '{DatabaseName}' PostgreSQL database.", ex);
+            }
+        }
+
+        static bool IsConnectionOrLoginIssue(NpgsqlException x)
+        {
+            //TODO postgres - any other errors?
```

Comment:
Anything more we need to do here?

Reply:
I've just tested it, fixed it and removed the comment